src/ray/util/pipe_logger.cc

```diff
@@ -84,7 +84,11 @@ size_t Read(int read_fd, char *data, size_t len) {
   return bytes_read;
 }
 void CompleteWriteEOFIndicator(int write_fd) {
-  ssize_t bytes_written = write(write_fd, kEofIndicator.data(), kEofIndicator.length());
+  // Write a newliner to mark the end of application logging.
+  ssize_t bytes_written = write(write_fd, "\n", /*count=*/1);
```

Comment:
Can you tell me which test will fail if we rely on close instead of kEofIndicator? I want to debug it.

Reply:
My newly added one.

The issue is, if application dumps a "helloworld" string with no newliner, pipe logger read "<hello><EOF>" as a whole.

Reply:
OH, I mean the test that will fail if I remove EOFIndicator but just rely on read() == 0